### PR TITLE
Extract utility functions from endpoint

### PR DESCRIPTION
**Pull request by breeswish**
Extracted from PR #2860.

Please merge PR #2827 first.

### CODE REVIEW

**Comment by huachaohuang** [comment]
Can you squash your commits?

File: src/coprocessor/util.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use kvproto::coprocessor as coppb;
+use tipb::schema::ColumnInfo;
+
+use super::codec::datum::Datum;
+use super::codec::mysql;
+
+pub fn prefix_next(key: &[u8]) -> Vec<u8> {
```
Comment (AndreMouche) [inline src/coprocessor/util.rs]: Could we add some unit tests in this file?

Comment (hicqu) [reply src/coprocessor/util.rs]: And please improve this function to reduce alloc.

**Comment by hicqu** [comment]
rest LGTM.

File: src/coprocessor/util.rs (hunk 1 above)
Comment (breeswish) [reply src/coprocessor/util.rs]: @AndreMouche tests added, PTAL

File: src/coprocessor/util.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+}
+
+/// `is_point` checks if the key range represents a point.
+pub fn is_point(range: &coppb::KeyRange) -> bool {
+    range.get_end() == &*prefix_next(range.get_start())
+}
+
+#[inline]
+pub fn get_pk(col: &ColumnInfo, h: i64) -> Datum {
```
Comment (AndreMouche) [inline src/coprocessor/util.rs]: Any test for get_pk && is_point?

Comment (breeswish) [reply src/coprocessor/util.rs]: Their logic is obvious, I think it's not very important and can be merged into "improve test coverage" task.

**Comment by breeswish** [comment]
@Hoverbear Yes! :) We always squash our commits when the PR is merged.

**Pull request merged**